library/alloc/benches/vec_deque.rs

```diff
@@ -123,3 +123,31 @@ fn bench_extend_chained_bytes(b: &mut Bencher) {
         ring.extend(black_box(input1.iter().chain(input2.iter())));
     });
 }
+
+#[bench]
+fn bench_vec_deque_to_vec(b: &mut Bencher) {
+    let mut ring = VecDeque::from(vec![0; 1024]);
+    ring.shrink_to_fit();
+    assert_eq!(ring.len(), ring.capacity());
+
+    b.iter(move || {
+        assert_eq!(ring.capacity(), 1024);
```

Comment:
Benchmarks and tests should be separated. I assume this exists to assert that the allocation was reused. This can be better done by pointer casts in a test. There should already be some similar tests from the previous vec<->vecdeque conversion PRs.

Reply:
I mostly did this to avoid accidental UB if any of the conversions or `shrink_to_fit` ever changed their behavior, by preventing out of bounds memory accesses. Of course, in the current implementation this will never happen, but my idea was that it would be way easier to catch and debug with an `assert` if it ever changed. If this isn't wanted then I can of course remove the `assert`s.

Reply:
Ok, makes sense. Maybe add a safety comment to the unsafe blocks then that explains that they're guarded by the asserts.